jwst/wfs_combine/wfs_combine_step.py

```diff
@@ -31,46 +31,46 @@ def process(self, input_table):
-
-                # Get the list of science members in this pair
-                science_members = [
-                    member
-                    for member in which_set.members
-                    if member.exptype.lower() == 'science'
-                ]
-                infile_1 = science_members[0].expname
-                infile_2 = science_members[1].expname
-                outfile = which_set.name
-
-                # Create the step instance
-                wfs = wfs_combine.DataSet(
-                    infile_1, infile_2, outfile, self.do_refine, self.flip_dithers, self.psf_size,
-                    self.blur_size, self.n_size
-                )
-
-                # Do the processing
-                output_model = wfs.do_all()
-
-                # The DataSet class does not close its resources.  Do that here.
-                wfs.input_1.close()
-                wfs.input_2.close()
-
-                # Update necessary meta info in the output
-                output_model.meta.cal_step.wfs_combine = 'COMPLETE'
-                output_model.meta.asn.pool_name = asn_table.meta.asn_table.asn_pool
-                output_model.meta.asn.table_name = os.path.basename(input_table)
-                output_model.meta.filename = which_set.name
-
-                output_container.append(output_model)
+        asn_table = self.load_as_level3_asn(input_table)
+
+        self.log.info('Using input table: %s', input_table)
+        self.log.info('The number of pairs of input files: %g', len(asn_table['products']))
+
+        output_container = datamodels.ModelContainer()
+
+        # Process each pair of input images listed in the association table
+        for which_set in asn_table['products']:
+
+            # Get the list of science members in this pair
+            science_members = [
+                member
+                for member in which_set['members']
+                if member['exptype'].lower() == 'science'
+            ]
+            infile_1 = science_members[0]['expname']
+            infile_2 = science_members[1]['expname']
+            outfile = which_set['name']
+
+            # Create the step instance
+            wfs = wfs_combine.DataSet(
+                infile_1, infile_2, outfile, self.do_refine, self.flip_dithers, self.psf_size,
+                self.blur_size, self.n_size
+            )
+
+            # Do the processing
+            output_model = wfs.do_all()
+
+            # The DataSet class does not close its resources.  Do that here.
+            wfs.input_1.close()
+            wfs.input_2.close()
+
+            # Update necessary meta info in the output
+            output_model.meta.cal_step.wfs_combine = 'COMPLETE'
+            output_model.meta.asn.pool_name = asn_table['asn_pool']
+            output_model.meta.asn.table_name = os.path.basename(input_table)
+            output_model.meta.filename = which_set['name']
+
+            output_container.append(output_model)
```

Comment:
Because we're no longer using a "with" block here, should we explicitly do an `asn_table.close()` here or just assume it'll be taken care of by python when it goes out of scope?

Reply:
I copied my work from [here](https://github.com/spacetelescope/jwst/blob/master/jwst/pipeline/calwebb_coron3.py), and the asn is not explicitly closed, so I think it should be fine. I initially tried to just replace the datamodels.load() with the load_as... in the with block, but the method is not configured to work in such a way.